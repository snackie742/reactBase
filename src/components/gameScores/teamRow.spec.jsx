import React from 'react';
import { shallow } from 'enzyme';
import TeamRow from './teamRow';

const defaultProps = {
    city: 'St. Louis',
    score: '3',
    abbr: 'STL',
};

const setup = (propOverrides) => {
  const props = Object.assign({}, defaultProps, propOverrides);
  const wrapper = shallow(<TeamRow {...props} />);
  const component = wrapper.instance();
  return { wrapper, component };
};

describe('TeamRow', () => {
  describe('<TeamRow/>', () => {
    describe('(Snapshot)', () => {
      test('renders as expected with score', () => {
        const { wrapper } = setup(defaultProps);
        expect(wrapper).toMatchSnapshot();
      });
      test('renders as expected with time', () => {
        const propOverrides = {
            city: 'STL',
            abbr: 'STL',
            time: '1:00 PM',
        };
        const { wrapper } = setup(propOverrides);
        expect(wrapper).toMatchSnapshot();
      });
    });
  });
});


