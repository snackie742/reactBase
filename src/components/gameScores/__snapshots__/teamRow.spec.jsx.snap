// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TeamRow <TeamRow/> (Snapshot) renders as expected with score 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TeamRow
    abbr="STL"
    city="St. Louis"
    score="3"
    time=""
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="col-sm-2"
>
          <img
                    alt=""
                    id="clip"
                    src="STL.png"
          />
</div>,
        <div
          className="col-sm-8"
          style={
                    Object {
                              "textAlign": "left",
                            }
          }
>
          St. Louis
</div>,
        <div
          className="col-sm-2"
>
          3
</div>,
        "",
      ],
      "className": "row",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <img
            alt=""
            id="clip"
            src="STL.png"
/>,
          "className": "col-sm-2",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "",
            "id": "clip",
            "src": "STL.png",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "St. Louis",
          "className": "col-sm-8",
          "style": Object {
            "textAlign": "left",
          },
        },
        "ref": null,
        "rendered": "St. Louis",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "3",
          "className": "col-sm-2",
        },
        "ref": null,
        "rendered": "3",
        "type": "div",
      },
      "",
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="col-sm-2"
>
            <img
                        alt=""
                        id="clip"
                        src="STL.png"
            />
</div>,
          <div
            className="col-sm-8"
            style={
                        Object {
                                    "textAlign": "left",
                                  }
            }
>
            St. Louis
</div>,
          <div
            className="col-sm-2"
>
            3
</div>,
          "",
        ],
        "className": "row",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              alt=""
              id="clip"
              src="STL.png"
/>,
            "className": "col-sm-2",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "",
              "id": "clip",
              "src": "STL.png",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "St. Louis",
            "className": "col-sm-8",
            "style": Object {
              "textAlign": "left",
            },
          },
          "ref": null,
          "rendered": "St. Louis",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "3",
            "className": "col-sm-2",
          },
          "ref": null,
          "rendered": "3",
          "type": "div",
        },
        "",
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`TeamRow <TeamRow/> (Snapshot) renders as expected with time 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TeamRow
    abbr="STL"
    city="STL"
    score="3"
    time="1:00 PM"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="col-sm-2"
>
          <img
                    alt=""
                    id="clip"
                    src="STL.png"
          />
</div>,
        <div
          className="col-sm-8"
          style={
                    Object {
                              "textAlign": "left",
                            }
          }
>
          STL
</div>,
        <div
          className="col-sm-2"
>
          3
</div>,
        <div
          className="col-sm-6 col-lg-5"
          style={
                    Object {
                              "fontSize": "1.35rem",
                            }
          }
>
          1:00 PM
</div>,
      ],
      "className": "row",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <img
            alt=""
            id="clip"
            src="STL.png"
/>,
          "className": "col-sm-2",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "",
            "id": "clip",
            "src": "STL.png",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "STL",
          "className": "col-sm-8",
          "style": Object {
            "textAlign": "left",
          },
        },
        "ref": null,
        "rendered": "STL",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "3",
          "className": "col-sm-2",
        },
        "ref": null,
        "rendered": "3",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "1:00 PM",
          "className": "col-sm-6 col-lg-5",
          "style": Object {
            "fontSize": "1.35rem",
          },
        },
        "ref": null,
        "rendered": "1:00 PM",
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="col-sm-2"
>
            <img
                        alt=""
                        id="clip"
                        src="STL.png"
            />
</div>,
          <div
            className="col-sm-8"
            style={
                        Object {
                                    "textAlign": "left",
                                  }
            }
>
            STL
</div>,
          <div
            className="col-sm-2"
>
            3
</div>,
          <div
            className="col-sm-6 col-lg-5"
            style={
                        Object {
                                    "fontSize": "1.35rem",
                                  }
            }
>
            1:00 PM
</div>,
        ],
        "className": "row",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              alt=""
              id="clip"
              src="STL.png"
/>,
            "className": "col-sm-2",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "",
              "id": "clip",
              "src": "STL.png",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "STL",
            "className": "col-sm-8",
            "style": Object {
              "textAlign": "left",
            },
          },
          "ref": null,
          "rendered": "STL",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "3",
            "className": "col-sm-2",
          },
          "ref": null,
          "rendered": "3",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "1:00 PM",
            "className": "col-sm-6 col-lg-5",
            "style": Object {
              "fontSize": "1.35rem",
            },
          },
          "ref": null,
          "rendered": "1:00 PM",
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
